src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -0,0 +1,539 @@
+                    tasks.Add(Task.Run(() => GetExtensionMethodSymbolsFromProjectAsync(
+                        project,
+                        forceIndexCreation,
+                        cancellationToken), cancellationToken));
+                }
+
+                using var _2 = ArrayBuilder<IMethodSymbol>.GetInstance(out var symbols);
+                var isPartialResult = false;
+
+                var results = await Task.WhenAll(tasks).ConfigureAwait(false);
+
+                foreach (var result in results)
+                {
+                    // `null` indicates we don't have the index ready for the corresponding project/PE,
+                    // we will queue a background task to force creating them. Meanwhile, returns
+                    // what we do have even it means we only show partial results.
+                    if (result == null)
+                    {
+                        isPartialResult = true;
+                        continue;
+                    }
+
+                    symbols.AddRange(result);
+                }
+
+                return (symbols.ToImmutable(), isPartialResult);
+            }
+
+            private static ImmutableArray<Project> GetAllRelevantProjects(Project project)
+            {
+                var solution = project.Solution;
+                var graph = solution.GetProjectDependencyGraph();
+                var relevantProjectIds = graph.GetProjectsThatThisProjectTransitivelyDependsOn(project.Id).Concat(project.Id);
+                return relevantProjectIds.Select(id => solution.GetRequiredProject(id)).Where(p => p.SupportsCompilation).ToImmutableArray();
+            }
+
+            private static ImmutableArray<PortableExecutableReference> GetAllRelevantPeReferences(Project project)
+                => project.MetadataReferences.OfType<PortableExecutableReference>().ToImmutableArray();
+
+            private async Task<ImmutableArray<IMethodSymbol>?> GetExtensionMethodSymbolsFromProjectAsync(
+                Project project,
+                bool forceIndexCreation,
+                CancellationToken cancellationToken)
+            {
+                // By default, don't trigger index creation except for documents in current project.
+                var isCurrentProject = project == CurrentProject;
+                forceIndexCreation = forceIndexCreation || isCurrentProject;
+
+                var cacheEntry = await GetCacheEntryAsync(
+                    project, loadOnly: !forceIndexCreation, CacheService, cancellationToken).ConfigureAwait(false);
+
+                if (!cacheEntry.HasValue)
+                {
+                    // Returns null to indicate index not ready
+                    return null;
+                }
+
+                if (!cacheEntry.Value.ContainsExtensionMethod)
+                {
+                    return ImmutableArray<IMethodSymbol>.Empty;
+                }
+
+                var currentAssembly = SemanticModel.Compilation.Assembly;
```

Comment:
you might want to call these OriginatingSemanticModel and OriginatingAssembly.  i was confused a bit as to what was going on here as i thought this SemanticModel/Compilatino was the same as teh Project you are *currently* looking at.  When dealing with two separate ones, having good names help keeps things clear.